bin/ncp-diag

```diff
@@ -76,24 +75,57 @@ function is_port_open()
   # than 450 request p/second a month, unless my calculations are wrong.
   # Thank you :pray: from Victor-ray, S. https://github.com/ZendaiOwl
   local -r PORTURL="https://checkport.zendai.net.eu.org/check"
-  local TYPE="${1?}" RESULT=("")
+  local TYPE="${1?}" IPType
   # Checks both port 80 & 443 for IPv4/IPv6 and returns the result or [N/A] [N/A]
-  test "$TYPE" -eq 4 && RESULT=("$(curl --silent --max-time 4 --ipv4 "$PORTURL" | jq -r '."80",."443"' )")
-  test "$TYPE" -eq 6 && RESULT=("$(curl --silent --max-time 4 --ipv6 "$PORTURL" | jq -r '."80",."443"' )")
-  test "$TYPE" -eq 0 && RESULT=("[N/A]" "[N/A]")
-  echo -n "${RESULT[@]}"
+  if ! [[ "$TYPE" =~ ^(0|4|6)$ ]]
+  then
+    echo "Invalid type parameter: $TYPE" 2>&1
```

Comment:
This output redirection has no effect. Do you mean `>&2`, maybe?

Reply:
It redirects STDERR to a copy and then into STDOUT :) 

Thought it was something like that you wanted to achieve with this line you wrote as suggestions earlier 

`echo "Invalid paramter: '$TYPE'" >2`

I can change it to `>&2` if you want? 

There isn't really any need for a redirection there at all tho, as the return value will be 0 from echo either way, then the function will return the value 1 by the next line executed

Reply:
Thinking about it further, there is actually no need for any such echo statements inside there at all, since the result is only dependent on the if statements outside that function, which only sets it to open if it can find a value in the array that equals to "open", otherwise the result will be "closed". So I can just set a `return 1` there

Reply:
I would still keep the echo statement in case the function is reused in the future. Regarding the redirection:

echo prints to STDOUT by default. However, since you interpret the stdout of the function as return value, errors and logs should instead by printed to STDERR, hence the `>&2`.
Otherwise, if a caller doesn't validate the return code correctly, they might try to interprete the error message as port check result

Reply:
output redirection does not affect the return code, btw

Reply:
Oh I see, I'll add an `echo` statement to it again then

Reply:
Now there is one again